server/src/test/scala/org/http4s/server/middleware/CORSSpec.scala

```diff
@@ -57,6 +76,10 @@ class CORSSpec extends Http4sSpec {
       cors2.orNotFound(req).map(_.headers must contain(headerCheck _)).unsafeRun
     }
 
+    "Respond with 403 when origin is not valid" in {
+      val req = buildRequest("/bar").replaceAllHeaders(Header("Origin", "http://blah.com/"))
+      cors2.orNotFound(req).map((resp: Response) => resp.status.code == 403).unsafeRun
```

Comment:
This behavior does is not defined in the CORS spec, but as @haripriyamurthy and I discussed, seems like the right thing to do and has precedent in other implementations (e.g., Tomcat).